### DISCUSSION TITLE
$refresh is not updating my view after adding a new record in the DB.

### DISCUSSION THREAD

**Discussion by kamdjouduplex** [livewire/livewire#4256] [1. Help] [answered]
Hi there, please I have been struggle with this:
Please I need your help. 
**Scenario:** I have two components **ComponentOne** and **ComponentTwo**. 

I am using ComponentOne to show the form to collect data and store in the DB it's working fine. 
Then, ComponentTwo is used to list the data form the DB. 
At the manual refresh of the page the view of ComponentTwo will render well and show the list of data from the mount() method. 

**Now my issue is that**, I am trying to update the list directly after a single record is added to the DB. 
See my code:

```
//ComponentOne                              
 public function saveSection(){ //method saving the record
        Section::create([
            'title' => $this->_title,
            'slug'  =>  strtolower(str_replace(' ','-',$this->_title)),
            'objectif' => $this->_objectif,
            'course_id' => $this->_course,
            'category_id' => $this->_category
        ]);
        $this->emit('refreshComponent'); /*Note: I am emitting the refresh action*/
  }
```

```
//ComponentTwo                                                                                               
namespace App\Http\Livewire;
use Livewire\Component;
use App\Models\Section;

class SectionList extends Component
{

    public $cerated_sections;

    protected $listeners = ['refreshComponent' => '$refresh']; /*Note: activating the refresh*/

    public function mount(){
        $this->cerated_sections = Section::all();
    }

    public function render()
    {
        return view('livewire.section-list');
    }
}
```

In the main page view I am having this:
```
<x-base-layout>
    <div class="max-w-7xl mx-auto bg-white">
        <div class="p-8">
            <h3 class="font-bold text-2xl">Editer le programme</h3>
        </div>
        <hr>
        <div class="p-8">
            <livewire:component-one />
        </div>
        <div class="p-8" id="coursSectionWraper">
            <livewire:component-two />
        </div>
    </div>
</x-base-layout> 
```

Please help needed.

**Comment by joshhanley** [comment, marked as answer]
@kamdjouduplex in your SectionList component, just calling `$refresh` in response to the event won't actually get any more sections from the database. It will only display the sections you had loaded when you called `Section::all()` in mount.

So what you need to do is create a method that the listener triggers, and inside that method you need to do another database call to get all the sections again.

```php
protected $listeners = ['refreshComponent' => 'getSections'];

public function getSections() {
    $this->cerated_sections = Section::all();
}
```

Hope this helps!

**Reply by kamdjouduplex** [reply]
Wooow!
Just just work like a like it was soo easy.
Thank you very much for your answer.

**Reply by joshhanley** [reply]
@kamdjouduplex no worries! 🙂

**Reply by junelzatura22** [reply]
It really works. Thank you.

**Reply by OwenMaddox** [reply]
Thank you so much I have been stuck on this for several hours!

**Comment by Muqoffin** [comment]
@joshhanley I have the same problem, and have tried the above solutions still not working, can you help me? i tried to install laravel jetstream and break the navigation-menu into several parts like sidebar and navbar but it causes refresh not running.

```
// component one
public function updateProfileInformation(UpdatesUserProfileInformation $updater)
{
    $this->resetErrorBag();
    $updater->update(
        Auth::user(),
        $this->photo
            ? array_merge($this->state, ['photo' => $this->photo])
            : $this->state
    );
    if (isset($this->photo)) {
        return redirect()->route('profile.show');
    }
    $this->emit('saved');
    $this->emit('refresh-navbar-menu');
    // $this->emitTo('refresh-navbar-menu');
    // $this->emitSelf('refresh-navbar-menu');
}
```


```
// component two
<?php

namespace App\Http\Livewire\Backend\Layouts;

use Livewire\Component;

class NavbarMenu extends Component
{
    /**
     * The component's listeners.
     *
     * @var array
     */
    protected $listeners = [
        'refresh-navbar-menu' => '$refresh',
    ];

    /**
     * Render the component.
     *
     * @return \Illuminate\View\View
     */
    public function render()
    {
        return view('livewire.backend.layouts.navbar-menu');
    }
}
```

```
// main
<div class="layout-wrapper layout-content-navbar  ">
  <div class="layout-container">
      <!-- Menu -->
      @livewire('backend.layouts.sidebar-menu')
      <!-- / Menu -->

      <!-- Layout container -->
      <div class="layout-page">
          <!-- Navbar -->
          @livewire('backend.layouts.navbar-menu')
          <!-- / Navbar -->

          <!-- Content wrapper -->
          <div class="content-wrapper">
              <!-- Content -->
              {{ $slot }}
              <!-- / Content -->

              <!-- Footer -->
              @livewire('backend.layouts.footer-menu')
              <!-- / Footer -->

              <div class="content-backdrop fade"></div>
          </div>
          <!-- Content wrapper -->
      </div>
      <!-- / Layout page -->
  </div>

  <!-- Overlay -->
  <div class="layout-overlay layout-menu-toggle"></div>
  <!-- Drag Target Area To SlideIn Menu On Small Screens -->
  <div class="drag-target"></div>

</div>
```

**Reply by joshhanley** [reply]
@Muqoffin when you say `$refresh` isn't running, how are you determining that? What exactly should change when refresh is called?

I would do a test to ensure that the event is being received correctly by implementing what I suggested above, mapping event to a method instead of `$refresh`, and then putting a `dump('working');` or something in it, so you can ensure event communication is actually happening between your components.

That aside, personally I wouldn't split my main layout into Livewire components like you have. I would probably only do it for the piece that needed interactivity (like user profile dropdown or something). Or I would wrap everything in a single Livewire component. See here for details https://joshhanley.com.au/articles/how-to-structure-your-layout-file-for-livewire

Hope this helps!
